From 40c37e9b39599673715da8961d0499469100f000 Mon Sep 17 00:00:00 2001
From: Bogdan Hamciuc <bogdan.hamciuc@freescale.com>
Date: Tue, 8 Jan 2013 00:36:02 +0200
Subject: [PATCH 003/518] dpaa_eth: MTU checking is not required for IP
 reassembled frames

The IP reassembled packets are not to be limited to MTU size.

Signed-off-by: Madalin Bucur <madalin.bucur@freescale.com>
[Grabbed from the branch, LINUX_IR5.2.0, of
https://git.freescale.com/git-private/cgit.cgi/ppc/alu-b4860/linux.git.]
Signed-off-by: Tiejun Chen <tiejun.chen@windriver.com>
---
 drivers/net/ethernet/freescale/dpa/dpaa_eth_sg.c |    4 +++-
 1 files changed, 3 insertions(+), 1 deletions(-)

diff --git a/drivers/net/ethernet/freescale/dpa/dpaa_eth_sg.c b/drivers/net/ethernet/freescale/dpa/dpaa_eth_sg.c
index 63f0f63..555ddef 100644
--- a/drivers/net/ethernet/freescale/dpa/dpaa_eth_sg.c
+++ b/drivers/net/ethernet/freescale/dpa/dpaa_eth_sg.c
@@ -502,7 +502,9 @@ void __hot _dpa_rx(struct net_device *net_dev,
 
 	skb->protocol = eth_type_trans(skb, net_dev);
 
-	if (unlikely(dpa_check_rx_mtu(skb, net_dev->mtu))) {
+	/* IP Reassembled frames are allowed to be larger than MTU */
+	if (unlikely(dpa_check_rx_mtu(skb, net_dev->mtu) &&
+		!(fd_status & FM_FD_IPR))) {
 		percpu_priv->stats.rx_dropped++;
 		goto drop_bad_frame;
 	}
-- 
1.7.5.4

